kafka/main.go

```diff
@@ -0,0 +1,141 @@
+package kafka
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	batgo_kafka "github.com/brave-intl/bat-go/utils/kafka"
+	"github.com/brave-intl/challenge-bypass-server/server"
+	"github.com/rs/zerolog"
+	"github.com/segmentio/kafka-go"
+	"github.com/sirupsen/logrus"
+)
+
+var brokers []string
+
+type Processor func([]byte, string, *server.Server, *zerolog.Logger)
+
+type TopicMapping struct {
+	Topic       string
+	ResultTopic string
+	Processor   Processor
+	Group       string
+}
+
+func StartConsumers(server *server.Server, logger *zerolog.Logger) error {
+	env := os.Getenv("ENV")
+	if env == "" {
+		env = "staging"
+	}
+	logger.Info().Msg(fmt.Sprintf("Starting %s Kafka consumers", env))
+	topicMappings := []TopicMapping{
+		TopicMapping{
+			Topic:       "request.redeem.v1." + env + ".cbp",
+			ResultTopic: "result.redeem.v1." + env + ".cbp",
+			Processor:   SignedTokenRedeemHandler,
+			Group:       "cbpProcessors",
+		},
+		TopicMapping{
+			Topic:       "request.sign.v1." + env + ".cbp",
+			ResultTopic: "result.sign.v1." + env + ".cbp",
+			Processor:   SignedBlindedTokenIssuerHandler,
+			Group:       "cbpProcessors",
+		},
+	}
+	var topics []string
+	for _, topicMapping := range topicMappings {
+		topics = append(topics, topicMapping.Topic)
+	}
+
+	consumer := newConsumer(topics, "cbpProcessors", logger)
+	var (
+		failureCount = 0
+		failureLimit = 10
+	)
+	logger.Trace().Msg("Beginning message processing")
+	for {
+		// `ReadMessage` blocks until the next event. Do not block main.
+		msg, err := consumer.ReadMessage(context.Background())
```

Comment:
as discussed out of band, we will make a small change and use the explicit commit API to ensure events are successfully processed and not dropped